main.go

```diff
@@ -33,7 +33,13 @@ func main() {
 	}
 }
 
-func run(ctx context.Context) error {
+func run(ctx context.Context) (err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			err = fmt.Errorf("panic: %s", r)
+		}
```

Comment:
I think you need to add a:
 fmt.Printf("\n%s\n", err)
after line 39, so that you might catch that in stdout when container is running
AND  also possibly log the line as a log.Info() ?

Reply:
I tested it locally by forcing a nil pointer in the code (initially I added a log here), but then it was logging it twice, once here and once in the function that calls this (line 31), example of an output:
```
{
  "created_at": "2021-11-19T12:13:08.763839Z",
  "errors": [
    {
      "message": "panic: runtime error: invalid memory address or nil pointer dereference",
      "stack_trace": [
        {
          "file": "/Users/david/go/pkg/mod/github.com/!o!n!sdigital/log.go/v2@v2.0.9/log/log.go",
          "function": "github.com/ONSdigital/log.go/v2/log.Fatal",
          "line": 102
        },
        {
          "file": "/Users/david/gomod/src/github.com/ONSdigital/dp-cantabular-csv-exporter/main.go",
          "function": "main.main",
          "line": 31
        },
        {
          "file": "/usr/local/Cellar/go/1.17.1/libexec/src/runtime/proc.go",
          "function": "runtime.main",
          "line": 255
        },
        {
          "file": "/usr/local/Cellar/go/1.17.1/libexec/src/runtime/asm_amd64.s",
          "function": "runtime.goexit",
          "line": 1581
        }
      ]
    }
  ],
  "event": "fatal runtime error",
  "namespace": "dp-cantabular-csv-exporter",
  "severity": 0
}
make: *** [debug] Error 1
```